CoinDock-api/app/Models/V1/Wallet.php

```diff
@@ -13,14 +20,134 @@ class Wallet extends Model
+    public function satoshiToCrypt($coin, $sathosis)
+    {
+        $satsToCoins = array_keys(config('assets.accepted_coins'));
 
-}
+        return ($satsToCoins[$coin]['sats_to_crypt'] * $sathosis);
+    }
+
+    //function to check whether the response is in json or not 
+    public function isJson($string)
+    {
+        json_decode($string);
+        return (json_last_error() == JSON_ERROR_NONE);
+    }
+
+    //getting basePath for for checking Wallet Balance
+    public function basePath($userCoinId, $walletId)
+    {
+        $userCoinName = Coin::whereId($userCoinId)->first()->name;
+
+        $coinList = config('assets.accepted_coins');
+        $coinKeys =  array_keys($coinList);
+
+        $basePath = '';
+
+        foreach ($coinKeys as $coin) {
+            if ($userCoinName == $coin) {
+                $basePath  = config('assets.accepted_coins');
+                $basePath = $basePath[$coin]['bal_path'];
+            }
+        }
+
+        return $basePath = str_replace('{id}', $walletId, $basePath);
+    }
+
+    //Converting every crypto currency into USD
+    public function cryptoToUsd($coin)
+    {
+        $assetsConfig = config('assets.accepted_coins');
+        $assetShortName = $assetsConfig[$coin]['coin_id'];
+
+        $cryptConversionBasePath = config('assets.coin_api.base_path') . config('assets.coin_api.crypto_usd');
+        $cryptConversionPath = str_replace('{id}', $assetShortName, $cryptConversionBasePath);
+        $balanceInUsd = Http::withHeaders(['X-CoinAPI-Key' => config('assets.coin_api.key')])
+            ->get($cryptConversionPath)['rate'];
+
+        return $balanceInUsd;
+    }
+
+    //Fetching number of coins through the Response
+    public function coins($response, $coin)
+    {
+        $responseArray = json_decode($response, true);
+        $responseArrayKeys = array_keys($responseArray);
+
+        foreach ($responseArrayKeys as $jsonKey) {
+            if ($jsonKey == 'balance' || $jsonKey == 'data' || $jsonKey == 'result') {
+
+                $coins = $responseArray[$jsonKey];
+                return $this->coinCheck($coin, $coins);
+            } elseif ($jsonKey == 'confirmed') {
+
+
+                $confirmedResponse = json_decode($response, true);
+                $confirmedResponse = json_encode($confirmedResponse['confirmed'], true);
+                $confirmedResponseArray = json_decode($confirmedResponse, true);
+
+                $coins = $confirmedResponseArray['nanoErgs'];
+                return $this->coinCheck($coin, $coins);
+            }
+        }
+    }
 
+    //Adding Wallet for Particular User
+    public function addWallet(User $user,Request $request)
+    {
+        $walletId = $request->wallet_id;
+        $walletCheck = Wallet::whereWalletId($walletId)->first();
+
+        if ($walletCheck) {
+            return response([
+                'message' => 'Wallet Already Added'
+            ], Response::HTTP_CONFLICT);
+        }
+
```

Comment:
The above conditional check must be handle in controllers only.

Reply:
done